### PR TITLE
Integrated Rise and DPIA type and kinding systems

### PR DESCRIPTION
**Pull request by michel-steuwer**
This PR removed the `DataType` and `Kinds` from DPIA and uses the equivalent types from Rise.

### CODE REVIEW

**Review by Bastacyclop — COMMENTED** [review]
I think that it would be good to document the difference between `FragmentKind.Identifier` and `FragmentIdentifier`, both names look like synonyms to me. Maybe with a small comment on `FragmentKind.Identifier` explaining that it is a wrapper. Or maybe it could be named more appropriately (`AnyIdentifier.Fragment`?).

File: src/main/scala/rise/core/types/Solution.scala [hunk 1]
```diff
@@ -30,10 +31,10 @@ object Solution {
 }
 
 case class Solution(ts: Map[Type, Type],
-                    ns: Map[NatIdentifier, Nat],
+                    ns: Map[Nat, Nat],
```
[the remark this answers is not in the document]
Comment (michel-steuwer) [reply src/main/scala/rise/core/types/Solution.scala]: This was due to combining two substitution implementations which resulted in a change in the API.
I have reverted this API change.

File: src/main/scala/rise/autotune/package.scala [hunk 2]
```diff
@@ -28,7 +28,7 @@ package object autotune {
                    configFile: Option[String] = None,
                    hierarchicalHM: Boolean = false)
 
-  case class Sample(parameters: Map[NatIdentifier, Nat],
+  case class Sample(parameters: Map[Nat, Nat],
```
[the remark this answers is not in the document]
Comment (michel-steuwer) [reply src/main/scala/rise/autotune/package.scala]: Same as above, I have reverted this API change.

File: src/main/scala/rise/core/types/Type.scala [hunk 3]
```diff
@@ -17,147 +17,136 @@ final case class TypeIdentifier(name: String) extends Type {
...
-object f16 extends ScalarType { override def toString: String = "f16" }
-object f32 extends ScalarType { override def toString: String = "f32" }
-object f64 extends ScalarType { override def toString: String = "f64" }
-
-object NatType extends DataType { override def toString: String = "nat" }
-
-final case class OpaqueType(name: String) extends DataType {
-  override def toString: String = name
-}
+object DataType {
```
[the remark this answers is not in the document]
Comment (michel-steuwer) [reply src/main/scala/rise/core/types/Type.scala]: This is not necessary, but I think it is useful if we want to think about automatically generating the data type implementation (which I am) and then achieve appropriate encapsulation.

**Comment by michel-steuwer** [comment]
@Bastacyclop Thanks for the review. I have addressed your comments.

> I think that it would be good to document the difference between `FragmentKind.Identifier` and `FragmentIdentifier`, both names look like synonyms to me. Maybe with a small comment on `FragmentKind.Identifier` explaining that it is a wrapper. Or maybe it could be named more appropriately (`AnyIdentifier.Fragment`?).

I have renamed the nested `*Kind.Identifier` classes to `*Kind.IDWrapper` to clarify their use.
I think we might be able to revisit this design when we move to Scala 3 and can use a union type to describe all possible variations of identifiers instead of these wrappers.

**Review by Bastacyclop — APPROVED** [review]

**Pull request merged**